hunts/models.py

```diff
@@ -47,27 +48,72 @@ def get_num_unlocked(self):
     def get_num_metas_solved(self):
         return self.puzzles.filter(Q(status=Puzzle.SOLVED), Q(is_meta=True)).count()
 
-    def get_solves_per_hour(self):
-        current_time = timezone.now()
+    def get_num_metas_unsolved(self):
+        return self.puzzles.filter(~Q(status=Puzzle.SOLVED), Q(is_meta=True)).count()
+
+    # Gets the number of puzzles that are either solved or have a solved meta via DFS.
+    def get_progression(self):
+        solved_puzzles = self.puzzles.filter(status=Puzzle.SOLVED)
+        solved_puzzle_ids = {p.id for p in solved_puzzles}
+        puzzles_to_check = [p for p in solved_puzzles]
+
+        while len(puzzles_to_check) > 0:
+            next_puzzle = puzzles_to_check.pop()
+            if next_puzzle.is_meta:
+                for feeder in next_puzzle.feeders.all():
+                    if feeder.id not in solved_puzzle_ids:
+                        solved_puzzle_ids.add(feeder.id)
+                        puzzles_to_check.append(feeder)
+
```

Comment:
Would this get all puzzles below a solved puzzle in the "puzzle tree" as well? (Including a feeder of that puzzle's feeder, e.g.)

Reply:
Upon further thought, I think I'm just going to go with your suggestion. If there is a difference, I can't see it mattering until very late in the hunt (when we've solved the final metameta/puzzle in the puzzlehunt or something), so I'm just going to go with "is solved or has a meta that's solved".

Reply:
Ah I see, I missed that you were traversing all the way down the tree. 

I think this is similar to #322 in that I think ideally both want the same set of puzzles (every "unnecessary" puzzle, where a puzzle is unnecessary if it is solved or if all its metas are unnecessary). 

I think the right way to do this in the DB is with a recursive CTE, something like (pseudocode):

```
WITH RECURSIVE unnecessary_puzzles (id, metas) AS (SELECT id, metas FROM puzzles WHERE puzzle is solved UNION ALL SELECT id, metas FROM puzzles WHERE metas in unnecessary_puzzles) SELECT * FROM unnecessary_puzzles```

though idk if we want to go down that path or not. Anyway what you did is fine, I think you can alternatively prefetch `feeders` in your initial query and it should prefetch everything into one query

Reply:
I got it to work with the following recursive call:
```
WITH RECURSIVE progression_puzzles (id) AS (
  SELECT id
  FROM puzzles_puzzle
  WHERE status = 'SOLVED' AND hunt_id = %d
  UNION
  (
    -- workaround psql error: "recursive reference in a subquery"
    WITH progression_ids AS (
        SELECT id FROM progression_puzzles
    )
    SELECT Metas.from_puzzle_id
    FROM puzzles_puzzle_metas Metas
    WHERE Metas.to_puzzle_id IN (SELECT * FROM progression_ids)
  )
)
SELECT id FROM progression_puzzles
```
This might be the hackiest thing I've done all week.

Reply:
Update: This doesn't work with SQLite (which Github's currently using to test), so I used an `INNER JOIN` instead.
```
WITH RECURSIVE progression_puzzles (id) AS (
            SELECT id
            FROM puzzles_puzzle
            WHERE (status = 'SOLVED' AND hunt_id = %s)
            UNION
            SELECT Metas.from_puzzle_id
            FROM puzzles_puzzle_metas Metas
            INNER JOIN progression_puzzles ON progression_puzzles.id = Metas.to_puzzle_id
        )
        SELECT id FROM progression_puzzles
```